nodebuilder/tests/sync_test.go

```diff
@@ -17,60 +17,99 @@ import (
+	numBlocks = 20
+	bsize     = 16
+	btime     = time.Millisecond * 300
 )
 
 /*
-Test-Case: Sync a Light Node with a Bridge Node(includes DASing of non-empty blocks)
+Test-Case: Header and block/sample sync against a Bridge Node of non-empty blocks.
+
 Steps:
 1. Create a Bridge Node(BN)
 2. Start a BN
 3. Check BN is synced to height 20
-4. Create a Light Node(LN) with a trusted peer
+
+Light node:
+4. Create a Light Node (LN) with bridge as a trusted peer
 5. Start a LN with a defined connection to the BN
-6. Check LN is synced to height 30
+6. Check LN is header-synced to height 20
+7. Wait until LN has sampled height 20
+8. Wait for LN DASer to catch up to network head
+
+Full node:
+4. Create a Full Node (FN) with bridge as a trusted peer
+5. Start a FN with a defined connection to the BN
+6. Check FN is header-synced to height 20
+7. Wait until FN has synced block at height 20
+8. Wait for FN DASer to catch up to network head
 */
-func TestSyncLightWithBridge(t *testing.T) {
+func TestSyncAgainstBridge(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), swamp.DefaultTestTimeout)
 	t.Cleanup(cancel)
 
 	sw := swamp.NewSwamp(t, swamp.WithBlockTime(btime))
-	fillDn := swamp.FillBlocks(ctx, sw.ClientContext, sw.Accounts, bsize, blocks)
+	// wait for core network to fill 20 blocks
+	fillDn := swamp.FillBlocks(ctx, sw.ClientContext, sw.Accounts, bsize, numBlocks)
+	sw.WaitTillHeight(ctx, numBlocks)
 
+	// start a bridge and wait for it to sync to 20
 	bridge := sw.NewBridgeNode()
-
-	sw.WaitTillHeight(ctx, 20)
-
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
-
-	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
-	require.NoError(t, err)
-
-	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 20))
-
-	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
-	require.NoError(t, err)
-
-	cfg := nodebuilder.DefaultConfig(node.Light)
-	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
-	light := sw.NewNodeWithConfig(node.Light, cfg)
-
-	err = light.Start(ctx)
-	require.NoError(t, err)
-
-	h, err = light.HeaderServ.GetByHeight(ctx, 30)
-	require.NoError(t, err)
-
-	err = light.ShareServ.SharesAvailable(ctx, h.DAH)
-	assert.NoError(t, err)
-
-	err = light.DASer.WaitCatchUp(ctx)
+	h, err := bridge.HeaderServ.GetByHeight(ctx, numBlocks)
 	require.NoError(t, err)
-
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 30))
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, numBlocks))
+
+	t.Run("light sync against bridge", func(t *testing.T) {
+		// create a light node that is connected to the bridge node as
+		// a bootstrapper
+		cfg := nodebuilder.DefaultConfig(node.Light)
+		addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+		require.NoError(t, err)
+		cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+		light := sw.NewNodeWithConfig(node.Light, cfg)
```

Comment:
This really asks to be hidden away

Reply:
https://github.com/celestiaorg/celestia-node/pull/2162/commits/0c19eb09514b56d1dad13d4ac251270cf9c2f2a1